packages/flutter/lib/src/animation/animation_controller.dart

```diff
@@ -515,69 +520,23 @@ class AnimationController extends Animation<double>
-          'The "duration" or "reverseDuration" property should be set, either in the constructor or later, before '
-          'calling the reverse() function.',
-        );
-      }
-      return true;
-    }());
-    assert(
-      _ticker != null,
-      'AnimationController.reverse() called after AnimationController.dispose()\n'
-      'AnimationController methods should not be used after calling dispose.',
-    );
-    _direction = _AnimationDirection.reverse;
-    if (from != null) {
-      value = from;
-    }
-    return _animateToInternal(lowerBound);
-  }
+  TickerFuture reverse({ double? from }) => _toggle('reverse', false, from);
 
-  /// Toggles the direction of this animation, based on whether it [isForwardOrCompleted].
+  /// Toggles the direction of this animation.
   ///
-  /// Specifically, this function acts the same way as [reverse] if the [status] is
-  /// either [AnimationStatus.forward] or [AnimationStatus.completed], and acts as
-  /// [forward] for [AnimationStatus.reverse] or [AnimationStatus.dismissed].
+  /// The value of `forward` should toggle between `true` and `false`, and
+  /// `toggle()` should be called each time it changes. This function acts
+  /// like `forward()` or `reverse()` based on whether the value passed is
+  /// `true` or `false`, respectively.
   ///
-  /// If [from] is non-null, it will be set as the current [value] before running
-  /// the animation.
+  /// If no argument is provided, the controller changes direction based on
+  /// the current [status].
   ///
   /// The most recently returned [TickerFuture], if any, is marked as having been
   /// canceled, meaning the future never completes and its [TickerFuture.orCancel]
   /// derivative future completes with a [TickerCanceled] error.
-  TickerFuture toggle({ double? from }) {
-    assert(() {
-      Duration? duration = this.duration;
-      if (isForwardOrCompleted) {
-        duration ??= reverseDuration;
-      }
-      if (duration == null) {
-        throw FlutterError(
-          'AnimationController.toggle() called with no default duration.\n'
-          'The "duration" property should be set, either in the constructor or later, before '
-          'calling the toggle() function.',
-        );
-      }
-      return true;
-    }());
-    assert(
-      _ticker != null,
-      'AnimationController.toggle() called after AnimationController.dispose()\n'
-      'AnimationController methods should not be used after calling dispose.',
-    );
-    _direction = isForwardOrCompleted ? _AnimationDirection.reverse : _AnimationDirection.forward;
-    if (from != null) {
-      value = from;
-    }
-    return _animateToInternal(switch (_direction) {
-      _AnimationDirection.forward => upperBound,
-      _AnimationDirection.reverse => lowerBound,
-    });
+  TickerFuture toggle({ bool? forward }) {
+    return _toggle('toggle', forward ?? !isForwardOrCompleted, null);
```

Comment:
Nit: You could also assign `forward` the default value of `!isForwardOrCompleted`.

Reply:
That'd be great, though I believe that default arguments need to be `const` values.